tools/tsp-client/src/commands.ts

```diff
@@ -480,13 +480,50 @@ export async function generateConfigFilesCommand(argv: any) {
   if (Object.keys(overrideJson).length > 0) {
     emitterPackageJson["overrides"] = overrideJson;
   }
-  
-  const emitterPath = resolveEmitterPathFromArgs(argv) ?? joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
-  
-  await writeFile(
-    emitterPath,
-    JSON.stringify(emitterPackageJson, null, 2),
-  );
+
+  const emitterPath =
+    resolveEmitterPathFromArgs(argv) ??
+    joinPaths(await getRepoRoot(outputDir), defaultRelativeEmitterPackageJsonPath);
+
+  try {
+    const existingEmitterPackageJson = JSON.parse(await readFile(emitterPath, "utf8"));
+    // If there's an existing emitter-package.json, we need to merge any untracked devDependencies
+    if (existingEmitterPackageJson) {
+      const currentDevDependencies = existingEmitterPackageJson["devDependencies"] ?? {};
+      const packageLockPath = joinPaths(dirname(packageJsonPath), "package-lock.json");
+      let existingPackageLockJson;
+      try {
+        // Check if package-lock.json exists
+        await stat(packageLockPath);
+        existingPackageLockJson = JSON.parse(await readFile(packageLockPath, "utf8"));
```

Comment:
is there a chance that the lock file is invalid, this could log a confusing error then.

Also I think as you doing a whole catch you can just catch the read(remove the stat)

Reply:
Could be, I updated the message we log to be more generic. Also removed the stat call